src/vector/geometry.rs

```diff
@@ -341,6 +343,152 @@ impl Geometry {
             gdal_sys::OGR_G_AssignSpatialReference(self.c_geometry(), spatial_ref.to_c_hsrs())
         };
     }
+
+    /// Computes and returns the bounding envelope for this geometry
+    pub fn envelope(&self) -> gdal_sys::OGREnvelope {
+        let mut envelope = OGREnvelope {
+            MinX: 0.0,
+            MaxX: 0.0,
+            MinY: 0.0,
+            MaxY: 0.0,
+        };
+        unsafe { gdal_sys::OGR_G_GetEnvelope(self.c_geometry(), &mut envelope) };
+        envelope
+    }
+
+    /// Computes and returns the bounding envelope (3D) for this geometry
+    pub fn envelope3d(&self) -> gdal_sys::OGREnvelope3D {
+        let mut envelope = OGREnvelope3D {
+            MinX: 0.0,
+            MaxX: 0.0,
+            MinY: 0.0,
+            MaxY: 0.0,
+            MinZ: 0.0,
+            MaxZ: 0.0,
+        };
+        unsafe { gdal_sys::OGR_G_GetEnvelope3D(self.c_geometry(), &mut envelope) };
+        envelope
+    }
+
+    /// Determines whether two geometries intersect.
+    /// If GEOS is enabled, then this is done in rigorous fashion otherwise TRUE is returned if the envelopes (bounding boxes)
+    /// of the two geometries overlap.
+    pub fn intersects(&self, other: &Geometry) -> bool {
+        unsafe { gdal_sys::OGR_G_Intersects(self.c_geometry(), other.c_geometry()) == 1 }
```

Comment:
Missing error checking.

Reply:
This method does net set error if geos is not enabled but return a degraded test result (see https://github.com/OSGeo/gdal/blob/master/gdal/ogr/ogrgeometry.cpp#L503)

Reply:
Ah, all right -- might be worth adding a comment for people less familiar with GDAL's API.

Reply:
For symetry reason, we may also return a `Result<bool>` but the return value will always be `Ok(_)`.